### PR TITLE
libobs: Fix obs_parse_avc_header missing high profile parameters

### PR DESCRIPTION
**Pull request by derrod**
### Description

Adds extended information for high* profiles to `obs_parse_avc_header()`.

### Motivation and Context

When testing #10608 I received a warning from a third party tool that the `AVCDecoderConfigurationRecord` produced by `obs_parse_avc_header()` was missing these things.

ISO/IEC 14496-15 Section 5.3.3.1.2 says this should be present for high/high10/high422/high444 profiles (high and high444 are supported by OBS).

Given that nobody cared for the past 10 years this is not critical, but still nice to be correct.

### How Has This Been Tested?

Recorded a video with #10608 and validated the data is now present.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: libobs/obs-avc.c [hunk 1]
```diff
@@ -202,6 +256,26 @@ size_t obs_parse_avc_header(uint8_t **header, const uint8_t *data, size_t size)
 	s_wb16(&s, (uint16_t)pps_size);
 	s_write(&s, pps, pps_size);
 
+	uint8_t profile_idc = sps[1];
+
+	/* Additional data required for high, high10, high422, high444 profiles.
+	 * See ISO/IEC 14496-15 Section 5.3.3.1.2 */
```
Comment (RytoEX) [inline libobs/obs-avc.c:262]:
```suggestion
	 * See ISO/IEC 14496-15 Section 5.3.3.1.2. */
```

File: libobs/obs-avc.c [hunk 2]
```diff
@@ -202,6 +256,26 @@ size_t obs_parse_avc_header(uint8_t **header, const uint8_t *data, size_t size)
...
+	if (profile_idc == 100 || profile_idc == 110 || profile_idc == 122 ||
+	    profile_idc == 244) {
+		uint8_t chroma_format_idc, bit_depth_luma, bit_depth_chroma;
```
Comment (RytoEX) [inline libobs/obs-avc.c:265]:
```suggestion
		uint8_t chroma_format_idc;
		uint8_t bit_depth_luma;
		uint8_t bit_depth_chroma;
```

**Review by RytoEX — CHANGES_REQUESTED** [review]
Nits.

File: libobs/obs-avc.c (hunk 2 above)
Comment (derrod) [reply libobs/obs-avc.c:265]: We don't break up declarations of the same type unless they also contain assignments. And I'd prefer to not do that.

Comment (RytoEX) [reply libobs/obs-avc.c:265]: I prefer distinct lines, but that's personal preference. Looks like the style guide permits this if there are no assignments.

**Review by RytoEX — APPROVED** [review]
Seems fine.

**Pull request merged**